### PR TITLE
Update to Turbolinks 5.0.0 final

### PR DESCRIPTION
**Pull request by sstephenson**

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Thank you Sam! I was doing this right now but I can't access RubyGems API :(

File: railties/lib/rails/generators/app_base.rb [hunk 1]
```diff
@@ -324,7 +324,7 @@ def javascript_gemfile_entry
                                        "Use #{options[:javascript]} as the JavaScript library")
 
           unless options[:skip_turbolinks]
-            gems << GemfileEntry.version("turbolinks", "~> 5.x",
+            gems << GemfileEntry.version("turbolinks", "~> 5",
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/app_base.rb]: Do we want to people upgrade to 6 without being careful? I prefer a more conservative version like `~> 5.0`

**Comment by dhh** [comment]
I think it's fine that someone could get 6. We'll make sure to coordinate a TL6 release in such a way that it's compatible with Rails 5 or barking in the right way.

**Comment by sstephenson** [comment]
Hmm, I thought `~> 5` meant all 5.* versions, but not 6. That’s what I intended.

http://stackoverflow.com/questions/24121960/why-does-the-gemfile-semantic-versioning-operator-produce-inconsistent-resu

**Comment by rafaelfranca** [comment]
Oh! TIL.

**Comment by rafaelfranca** [comment]
Backported in 9ec5241 and fbbfb78

**Pull request merged**